app/frontend/src/components/Markdown.tsx

```diff
@@ -1,26 +1,77 @@
-import { Box, BoxProps, makeStyles } from "@material-ui/core";
-import MarkdownIt from "markdown-it";
-import React from "react";
+import "@toast-ui/editor/dist/toastui-editor-viewer.css";
 
-const md = new MarkdownIt().disable("image");
+import { makeStyles } from "@material-ui/core";
+import ToastUIViewer from "@toast-ui/editor/dist/toastui-editor-viewer";
+import { useEffect, useRef } from "react";
 
-interface MarkdownProps extends BoxProps {
+interface MarkdownProps {
   source: string;
+  topHeaderLevel?: number;
 }
 
 const useStyles = makeStyles((theme) => ({
   root: {
+    fontSize: theme.typography.fontSize,
     fontFamily: theme.typography.fontFamily,
+    "& h1": {
+      borderBottom: "none",
+      ...theme.typography.h1,
+    },
+    "& h2": {
+      borderBottom: "none",
+      ...theme.typography.h2,
+    },
+    "& h3": theme.typography.h3,
+    "& h4": theme.typography.h4,
+    "& h5": theme.typography.h5,
+    "& h6": theme.typography.h6,
+    "& p": theme.typography.body1,
+    "& ol": theme.typography.body1,
+    "& ul": theme.typography.body1,
+    "& blockquote": theme.typography.body1,
+    "& a": {
+      color: theme.palette.primary.main,
+    },
   },
 }));
 
-export default function Markdown({ source, ...otherProps }: MarkdownProps) {
+export default function Markdown({
+  source,
+  topHeaderLevel = 2,
+}: MarkdownProps) {
   const classes = useStyles();
-  return (
-    <Box
-      className={classes.root}
-      dangerouslySetInnerHTML={{ __html: md.render(source) }}
-      {...otherProps}
-    />
-  );
+
+  const rootEl = useRef<HTMLDivElement>(null);
+  const viewer = useRef<ToastUIViewer>();
+  useEffect(() => {
+    let sanitizedSource = increaseMarkdownHeaderLevel(source, topHeaderLevel);
+    //remove all html except <br>
+    sanitizedSource = sanitizedSource.replace(/<(?!br)([^>]+)>/gi, "");
+    //change images ![]() to links []()
+    sanitizedSource = sanitizedSource.replace(/!(?!\[.*\]\(.*\))/gi, "");
```

Comment:
Stripping html here except <br>. I didn't want to do it in the backend, because then you have to keep track of which fields are markdown. And you can't do it in the editor, because the client isn't trustworthy. So viewer it is.